lib/screens/onboarding/claim_code.dart

```diff
@@ -95,77 +117,125 @@ class _ClaimCodeState extends State<ClaimCode> {
 
   @override
   Widget build(BuildContext context) {
-    return Padding(
-      padding: const EdgeInsets.symmetric(horizontal: 33),
-      child: Column(
-        mainAxisAlignment: MainAxisAlignment.start,
-        children: <Widget>[
-          ClipboardTextField(
-            controller: inviteCodeController,
-            labelText: "Invite code (5 words)".i18n,
-            hintText: "Paste from clipboard".i18n,
-            onChanged: findInvite,
-          ),
-          SizedBox(height: 16),
-          status == ClaimCodeStatus.emptyInviteCode
-              ? Center(
-                  child: Text(
-                    "If you received invite from another Seeds member - enter secret words and it will be claimed automatically".i18n,
-                    style: TextStyle(
-                      fontSize: 14,
-                      fontFamily: "worksans",
-                      fontWeight: FontWeight.w300,
-                    ),
-                  ),
-                )
-              : Container(),
-          status == ClaimCodeStatus.searchingInvite
-              ? Center(
-                  child: Column(
-                    children: <Widget>[
-                      CircularProgressIndicator(),
-                      SizedBox(height: 5),
-                      Text("Looking for invite...".i18n),
+    final ScannerBloc scannerBloc = Provider.of(context);
+    scannerBloc.execute(SetTextControllerCmd(inviteCodeController, ScanContentType.inviteCode));
+
```

Comment:
Question - if scannerBloc is a global variable of sorts - there's only 1 instance in the system - then setting the controller on this would affect all other components that use it?

As a matter of policy, private variables should not go on global variables. So this could use its own scannerbloc instance instead of a global one?

So if the scanner bloc is tied to this component or stores parts of this widget then it needs to be its own instance.

Reply:
👍 

TextControllers are an annoyance when you don't only want to listen to them, but also update them. This is an experiment by me trying out different ways of feeding back the new text into the controller. I'll go back to the ordinary listener instead of this contrived stuff ;)

Reply:
I've returned to an ordinary listener now as you can see in the initState() method.

Reply:
Ok! :)